src/katgpucbf/xbgpu/xsend.py

```diff
@@ -21,26 +21,72 @@
 from collections.abc import Callable, Sequence
 from typing import Final
 
-import katsdpsigproc
 import katsdpsigproc.accel as accel
 import numpy as np
 import spead2
 import spead2.send.asyncio
+from katsdpsigproc.abc import AbstractContext
 from prometheus_client import Counter
 
 from .. import COMPLEX, DEFAULT_PACKET_PAYLOAD_BYTES
-from ..spead import FLAVOUR, FREQUENCY_ID, IMMEDIATE_FORMAT, TIMESTAMP_ID, XENG_RAW_ID
+from ..spead import FLAVOUR, FREQUENCY_ID, IMMEDIATE_FORMAT, TIMESTAMP_ID, XENG_RAW_ID, make_immediate
 from . import METRIC_NAMESPACE
 
 output_heaps_counter = Counter("output_x_heaps", "number of X-engine heaps transmitted", namespace=METRIC_NAMESPACE)
 output_bytes_counter = Counter(
     "output_x_bytes", "number of X-engine payload bytes transmitted", namespace=METRIC_NAMESPACE
 )
+output_visibilities_counter = Counter(
+    "output_x_visibilities", "number of scalar visibilities that saturated", namespace=METRIC_NAMESPACE
+)
+output_clipped_visibilities_counter = Counter(
+    "output_x_clipped_visibilities", "number of scalar visibilities that saturated", namespace=METRIC_NAMESPACE
+)
```

Comment:
```suggestion
output_visibilities_counter = Counter(
    "output_x_visibilities", "number of scalar visibilities", namespace=METRIC_NAMESPACE
)
output_clipped_visibilities_counter = Counter(
    "output_x_clipped_visibilities", "number of scalar visibilities that saturated", namespace=METRIC_NAMESPACE
)
```
copy-paste trouble perhaps?